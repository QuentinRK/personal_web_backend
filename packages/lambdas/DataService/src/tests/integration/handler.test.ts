/* eslint-disable no-undef */
import * as AWS from 'aws-sdk-mock';
import { APIGatewayProxyEvent } from 'aws-lambda';
import lambdaHandler from '../../main/Handler/index';
import projects from '../mocks/projects.json';
import bio from '../mocks/bio.json';

describe('Tests how lambda handler executes different routes', () => {
  beforeAll(() => {
    AWS.mock('DynamoDB', 'getItem', bio);
    AWS.mock('DynamoDB', 'query', projects);
  });

  it('verifies /bio endpoint data', async () => {
    const event: APIGatewayProxyEvent = {
      path: '/bio',
    } as any;
    const result = await lambdaHandler(event);

    expect(result.statusCode).toEqual(200);
    expect(result.body).toMatchSnapshot();
  });

  it('verifies /projects endpoint data', async () => {
    const event: APIGatewayProxyEvent = {
      path: '/projects',
    } as any;
    const result = await lambdaHandler(event);

    expect(result.statusCode).toEqual(200);
    expect(result.body).toMatchSnapshot();
  });

  it('Should throw not found error', async () => {
    const event: APIGatewayProxyEvent = {
      path: '/invalid-route',
    } as any;
    const result = await lambdaHandler(event);

    expect(result.statusCode).toEqual(404);
    expect(result.body).toEqual('{"message":"Route Not Found"}');
  });

  afterAll(() => {
    AWS.restore('DynamoDB');
  });
});
