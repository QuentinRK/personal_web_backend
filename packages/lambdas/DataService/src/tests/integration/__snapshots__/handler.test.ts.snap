// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests how lambda handler executes different routes verifies /about-me endpoint data 1`] = `"{\\"name\\":\\"bio\\",\\"description\\":\\"The creativity and freedom that comes with software development is what ensnared me and motivated me to learn Python, Javascript and SQL. Through the development of the following projects (using Django, Flask, React etc) I have honed my skills in order to create intuitive user interfaces and efficient backend programs, ultimately becoming a better problem solver.\\"}"`;

exports[`Tests how lambda handler executes different routes verifies /projects endpoint data 1`] = `"[{\\"name\\":\\"calq\\",\\"description\\":\\"Also developed in PYQT5, CalQ is a calculator that can be used on a desktop. While developing CalQ, the main focus of this application was the frontend design. This project helped solidify my knowledge in Object Orientated Programming.\\",\\"link\\":\\"https://github.com/QuentinRK/CalcQ\\"},{\\"name\\":\\"esotrix\\",\\"description\\":\\"Eostrix is a web app that displays real-time cryptocurrency prices. This application utilises websocket connections to stream and display these prices. The server side of Eostrix was developed in Flask and the client side was created using Javascript.\\",\\"link\\":\\"http://www.eostrix.co.uk/\\"},{\\"name\\":\\"hunterx\\",\\"description\\":\\"HunterX is a CLI application that is used to find the best deals for products. HunterX takes in command line arguments such as the product of interest. This generates deals from different companies , it also gives users the options to create an account and save deals. HunterX was developed Python and Postgresql.\\",\\"link\\":\\"https://github.com/QuentinRK/HunterX\\"},{\\"name\\":\\"salsasecure\\",\\"description\\":\\"Salsa-Secure is a GUI application developed in PYQT5, a framework in Python. Salsa-Secure compresses directories into archived files, the bytes in the folder are then scrambled with the Salsa20 cypher. The generated file is encrypted via password and can be decrypted.\\",\\"link\\":\\"https://github.com/QuentinRK/Salsa-Secure\\"}]"`;
